test/js/node/url/url-domain-ascii-unicode.test.js

```diff
@@ -1,31 +1,31 @@
-import { describe, test } from "bun:test";
-import assert from "node:assert";
+import { describe, test, expect } from "bun:test";
 import url from "node:url";
 
-const domainToASCII = url.domainToASCII;
-const domainToUnicode = url.domainToUnicode;
+const pairs = [
+  ["ıíd", "xn--d-iga7r"],
+  ["يٴ", "xn--mhb8f"],
+  ["www.ϧƽəʐ.com", "www.xn--cja62apfr6c.com"],
+  ["новини.com", "xn--b1amarcd.com"],
+  ["名がドメイン.com", "xn--v8jxj3d1dzdz08w.com"],
+  ["افغانستا.icom.museum", "xn--mgbaal8b0b9b2b.icom.museum"],
+  ["الجزائر.icom.fake", "xn--lgbbat1ad8j.icom.fake"],
+  ["भारत.org", "xn--h2brj9c.org"],
+];
 
-// TODO: Support url.domainToASCII and url.domainToUnicode.
-describe.todo("url.domainToASCII and url.domainToUnicode", () => {
-  test("convert from unicode to ascii and back", () => {
-    const domainWithASCII = [
-      ["ıíd", "xn--d-iga7r"],
-      ["يٴ", "xn--mhb8f"],
-      ["www.ϧƽəʐ.com", "www.xn--cja62apfr6c.com"],
-      ["новини.com", "xn--b1amarcd.com"],
-      ["名がドメイン.com", "xn--v8jxj3d1dzdz08w.com"],
-      ["افغانستا.icom.museum", "xn--mgbaal8b0b9b2b.icom.museum"],
-      ["الجزائر.icom.fake", "xn--lgbbat1ad8j.icom.fake"],
-      ["भारत.org", "xn--h2brj9c.org"],
-    ];
+describe("url.domainToASCII", () => {
+  for (const [domain, ascii] of pairs) {
+    test(`convert from '${domain}' to '${ascii}'`, () => {
+      const domainConvertedToASCII = url.domainToASCII(domain);
+      expect(domainConvertedToASCII).toEqual(ascii);
+    });
+  }
+});
 
-    domainWithASCII.forEach(pair => {
-      const domain = pair[0];
-      const ascii = pair[1];
-      const domainConvertedToASCII = domainToASCII(domain);
-      assert.strictEqual(domainConvertedToASCII, ascii);
-      const asciiConvertedToUnicode = domainToUnicode(ascii);
-      assert.strictEqual(asciiConvertedToUnicode, domain);
+describe("url.domainToUnicode", () => {
+  for (const [domain, ascii] of pairs) {
+    test(`convert from '${ascii}' to '${domain}'`, () => {
+      const asciiConvertedToUnicode = url.domainToUnicode(ascii);
+      expect(asciiConvertedToUnicode).toEqual(domain);
     });
-  });
+  }
```

Comment:
Nice! These match Node's tests:
https://github.com/nodejs/node/blob/main/test/parallel/test-url-domain-ascii-unicode.js

Reply:
More tests can be found:
https://github.com/nodejs/node/blob/d96606d9aab35baf9e5c77cf9312145c3ae6086b/test/parallel/test-whatwg-url-custom-domainto.js
and
https://github.com/nodejs/node/blob/d96606d9aab35baf9e5c77cf9312145c3ae6086b/test/fixtures/url-idna.js

Reply:
thank you!

Reply:
adding these cases jumped the result from `16 pass 0 fail` to `111 pass 9 fail`

Reply:
fixed the remaining!